app/src/main/java/com/quarkworks/apartmentgroceries/MainActivity.java

```diff
@@ -1,55 +1,63 @@
 package com.quarkworks.apartmentgroceries;
 
 import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.AsyncTask;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
-import android.view.Menu;
-import android.view.MenuItem;
 
 import com.quarkworks.apartmentgroceries.auth.LoginActivity;
-import com.quarkworks.apartmentgroceries.auth.SignupActivity;
-import com.quarkworks.apartmentgroceries.user.UserActivity;
+import com.quarkworks.apartmentgroceries.main.HomeActivity;
 
 
 public class MainActivity extends AppCompatActivity {
 
-    private static final String LOG_TAG = MainActivity.class.getSimpleName();
+    private static final String TAG = MainActivity.class.getSimpleName();
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.main_activity);
+        new BackgroundSplashTask().execute();
     }
 
-    @Override
-    public boolean onCreateOptionsMenu(Menu menu) {
-        getMenuInflater().inflate(R.menu.main_menu, menu);
-        return true;
-    }
-
-    @Override
-    public boolean onOptionsItemSelected(MenuItem item) {
+    private class BackgroundSplashTask extends AsyncTask<Void, Void, Boolean> {
 
-        Intent intent;
-
-        switch (item.getItemId()) {
-            case R.id.action_login:
-                intent = new Intent(this, LoginActivity.class);
-                startActivity(intent);
-                return true;
-
-            case R.id.action_sign_up:
-                intent = new Intent(this, SignupActivity.class);
-                startActivity(intent);
-                return true;
+        @Override
+        protected void onPreExecute() {
+            super.onPreExecute();
+        }
 
-            case R.id.action_user:
-                intent = new Intent(this, UserActivity.class);
-                startActivity(intent);
-                return true;
+        @Override
+        protected Boolean doInBackground(Void... params) {
+            Boolean requireLogin = false;
+            SharedPreferences sharedPreferences = getApplication()
+                    .getSharedPreferences(getApplication().getString(R.string.login_or_sign_up_session), 0);
+            String sessionToken = sharedPreferences.getString("sessionToken", null);
+            if (sessionToken == null || sessionToken.isEmpty()) return true;
+
+            int SPLASH_SHOW_TIME = 3000;
+            try {
+                Thread.sleep(SPLASH_SHOW_TIME);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
```

Comment:
instead of doing these null checks you can simply use: return TextUtils.isEmpty(sessionToken);

TextUtils.isEmpty() returns true for null and length = 0 and false otherwise so you don't have to do a null check every time :)